qml/MainMenu.qml

```diff
@@ -166,20 +166,64 @@ Item {
                     enabled: !autoGainChB.checked
                     visible: SettingsManager.enableSensorAdvancedConfiguration
 
+                    property var maxLengthMm: SettingsManager.debugMode ? 1e6 : 5e4
+                    property var minLengthMm: 1e3
+
+                    function setDepth(startMm, lengthMm) {
+                        // Sensor max range is limited to maxLengthMm, this value is also the max window depth
+                        // The minimum range between start and length is minLengthMm
+
+                        // Start
+                        if(ping.start_mm !== startMm) {
+                            // Check if length + new start is bigger than our limit (maxLengthMm)
+                            if(startMm >= maxLengthMm - minLengthMm) {
+                                startMm = maxLengthMm - minLengthMm
+                            }
+
+                            // Limit window to maxLengthMm
+                            if(startMm + ping.length_mm > maxLengthMm) {
+                                lengthMm = maxLengthMm - startMm
+                            }
+
+                            // Set new value
+                            ping.start_mm = startMm
+                        }
+
+                        // Length
+                        if(ping.length_mm !== lengthMm) {
+                            // Check if length is less than minLengthMm
+                            var lengthMm = lengthMm
+                            if(lengthMm < minLengthMm) {
+                                lengthMm = minLengthMm
+                            }
+
+                            // Check if length is inside our maxLengthMm
+                            lengthMm = Math.min(lengthMm, maxLengthMm - ping.start_mm)
+
+                            // If length is invalid, set minLengthMm
+                            if(isNaN(lengthMm)) {
+                                lengthMm = minLengthMm
+                            }
+
+                            // Set new value
+                            ping.length_mm = lengthMm
+                        }
+                    }
+
+
+
                     PingTextField {
                         id: startLength
                         title: "Scan start point (mm):"
                         text: ping.start_mm
                         validator: IntValidator {
                             bottom: 0
-                            top: SettingsManager.debugMode ? 1e6 : 7e5
+                            top: maxDepthMm
```

Comment:
`[12:52:22.105] default[Warning]: MainMenu.qml(221) qrc:/MainMenu.qml:221: ReferenceError: maxDepthMm is not defined`